src/components/Checkbox.tsx

```diff
@@ -75,21 +22,64 @@ const Checkbox: React.FC<IProps> = ({
       callback()
     }
   }
+
+  const checkboxContainerAttributes = {
+    display: "inline-block",
+    verticalAlign: "middle",
+    className: className,
+    onClick: handleClick,
+  }
+
+  const hiddenCheckboxAttributes = {
+    type: "checkbox",
+    checked: checked,
+    readOnly: true,
+    ...rest,
+  }
+
+  const styledCheckboxAttributes = {
+    "aria-hidden": true,
+    className: "styled-checkbox",
+    display: "inline-block",
+    w: `${size}rem`,
+    h: `${size}rem`,
+    minW: `${size}rem`,
+    bg: `${checked ? "primary400" : "background"}`,
+    border: "1px",
+    borderStyle: "solid",
+    borderColor: "black50",
+    borderRadius: "3px",
+    transition: "all 150ms",
+    _hover: {
+      boxShadow: "tableItemBoxShadow",
+      border: "1px",
+      borderStyle: "solid",
+      borderColor: "primary600",
+      transition: "transform 0.1s",
+      transform: "scale(1.02)",
+    },
+  }
+
   return (
-    <CheckboxContainer className={className} onClick={handleClick}>
-      <HiddenCheckbox type="checkbox" checked={checked} readOnly {...rest} />
-      <StyledCheckbox
-        aria-hidden="true"
-        checked={checked}
-        className="styled-checkbox"
-        size={size}
-      >
-        <Icon checked={checked} viewBox="0 0 24 24">
+    <Box {...checkboxContainerAttributes}>
+      <VisuallyHiddenInput {...hiddenCheckboxAttributes} />
+      <Box {...styledCheckboxAttributes}>
```

Comment:
When using a screen reader, I'd expect the screen reader to announce that something is checked when it has been selected. Currently, it announces everything as 'unticked' in every state.

I think we should also pass `checked={checked}` to this Box to fix that.

Reply:
@minimalsm looking at the previous code, we were using the `checked` prop just for style matters. I don't think that would change anything else.

Reply:
@minimalsm @pettinarip I'm not sure how to proceed on this. I tried [this screen reader](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn) on the Gatsby preview build and it announces the checkbox as "ticked" if it is. Though I don't know if I'm testing it properly.

Looking at [this accessibility docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role) seems like we should use `role="checkbox"` and `aria-checked={checked}`.

Reply:
Since we are planning on using the Chakra native checkbox, we can skip this for now.

Reply:
@minimalsm let me know if you have any concerns about this before merging it.

Reply:
As chatted with @minimalsm we agree on this. We will cover these things on the implementation of the native Chakra checkbox.